resources/indexer.py

```diff
@@ -18,290 +19,287 @@
-        part = int(p.split('.')[0])
-
-    else:
-        raise Exception(f"Filename not in a known format, {file.name}.")
-
-    return year, run, subrun, part
-
-
-def _get_software(run_meta_xml):
-    """Return software metadata"""
-    softwares = []
-    for software in run_meta_xml['DIF_Plus']['Plus']['Project']:
-        soft_meta = {
-            'name': str(software['Name']),
-            'version': str(software['Version']),
-            'date': str(software['DateTime'])
-        }
-        softwares.append(soft_meta)
-    return softwares
+    def _parse_xml(self):
+        """Return data points from xml dict"""
+        if self.meta_xml:
+            start_dt = str(self.meta_xml['DIF_Plus']['Plus']['Start_DateTime'])
+            end_dt = str(self.meta_xml['DIF_Plus']['Plus']['End_DateTime'])
+            create_date = str(self.meta_xml['DIF_Plus']['DIF']['DIF_Creation_Date'])
+            software = self._get_software()
+        else:
+            start_dt = None
+            end_dt = None
+            create_date = date.fromtimestamp(os.path.getctime(self.file.path)).isoformat()
+            software = None
+
+        return start_dt, end_dt, create_date, software
+
+    def _get_software(self):
+        """Return software metadata"""
+
+        def parse_project(project):
+            software = {}
+            if 'Name' in project:
+                software['name'] = str(project['Name'])
+            if 'Version' in project:
+                software['version'] = str(project['Version'])
+            if 'DateTime' in project:
+                software['date'] = str(project['DateTime'])
+            return software
+
+        software_list = []
+        entry = self.meta_xml['DIF_Plus']['Plus']['Project']
+        entry_type = type(entry)
+
+        if entry_type is list:
+            for project in entry:
+                software_list.append(parse_project(project))
+        elif entry_type is collections.OrderedDict:
+            software_list = [parse_project(entry)]
+        else:
+            raise Exception(f"meta xml file has unanticipated 'Project' type {entry_type}.")
+
+        return software_list
+
+    def _get_events_data(self):
+        """Return the first/last event ids, number of events, and content status"""
+        first, last = None, None
+        count = 0
+        status = "good"
+        try:
+            for frame in dataio.I3File(self.file.path):
+                if 'I3EventHeader' in frame:
+                    count = count + 1
+                    event_id = int(frame['I3EventHeader'].event_id)
+                    if first is None or first > event_id:
+                        first = event_id
+                    if last is None or last < event_id:
+                        last = event_id
+        except:
+            status = "bad"
+        return first, last, count, status
+
+
+class L2FileMetadata(I3FileMetadata):
+    """Metadata for L2 i3 files"""
+    gaps_dict = dict()
+    gcd_filepath = ""
```

Comment:
You set these in the constructor, so I'd just drop these lines.

Reply:
Done.